folksonomy/api.py

```diff
@@ -361,13 +361,142 @@ async def product_tags_list(
 
     cur, timing = await db.db_exec(query, tuple(params))
     out = await cur.fetchone()
-    
+
     return JSONResponse(
         status_code=200,
         content=out[0] if out and out[0] is not None else [],
         headers={"x-pg-timing": timing}
     )
-    
+
+
+@app.get("/product/{product}/knowledge-panels", response_model=ProductKnowledgePanels)
+async def product_barcode_knowledge_panels(response: Response,
+                                           product: str,
+                                           k: Optional[str] = Query(None, description="Optional key to filter the results"),
+                                           owner='',
+                                           user: User = Depends(get_current_user)):
+    """
+    Return product information in a Knowledge Panel format.
+    If 'k' is provided, only that key is fetched.
+    If none is provided, all the keys are returned.
+    """
+    check_owner_user(user, owner, allow_anonymous=True)
+
+    query = f"""
+        SELECT product, k, v, owner, version, editor, last_edit, comment
+        FROM folksonomy
+        WHERE product = %s
+    """
+    params = [product]
+    if k:
+        query+= " AND k = %s"
+        params.append(k)
+    query += " ORDER BY k;"
+
+    cur, timing = await db.db_exec(query, tuple(params))
+    rows = await cur.fetchall()
+
+    if not rows:
+        raise HTTPException(
+                status_code=404,
+                detail="Could not find product or key",
+            )
```

Comment:
I would prefer that we reuse existing API functions here (or refactor them to use a common method).